packages/next-app/src/contexts/PoolMetadataContext.tsx

```diff
@@ -65,16 +66,20 @@ export function PoolMetadataProvider({ children }: { children: ReactNode }) {
     UpdateStatus.PINNING
   );
   const [submit, handleSubmit] = useState<boolean>(false);
+  const [metadataUpdated, setMetadataUpdated] = useState<boolean>(false);
```

Comment:
for instance, if the user changes one field, and then changes it back to the previous value (i.e. the one currently on IPFS) -- is `metadataUpdated` going to be true? 

I don't think that's a problem for now but we should log all improvements that still can be made on Linear

Reply:
Thanks for the catch!

I've updated the context to compare the original metadata state with the updated metadata. Now if we change a field and roll back to the same value, the update button will be disabled.